tests/snuba/test_snuba.py

```diff
@@ -122,18 +122,39 @@ def _query_for_issue(group_id):
         GroupHash.objects.create(
             project=self.project,
             group=group2,
-            hash=a_hash,
+            hash=hash,
         )
 
         # tombstone time is returned as expected
         assert snuba.get_project_issues([self.project], [group2.id]) == \
             [(group2.id, group2.project_id, [('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
-                           ght.deleted_at.strftime("%Y-%m-%d %H:%M:%S"))])]
+                                              ght.deleted_at.strftime("%Y-%m-%d %H:%M:%S"))])]
 
         # events <= to the tombstone date aren't returned
-        _insert_event_for_time(ght.deleted_at)
+        self._insert_event_for_time(ght.deleted_at, hash)
         assert _query_for_issue(group2.id) == {}
 
         # only the event > than the tombstone date is returned
-        _insert_event_for_time(ght.deleted_at + timedelta(seconds=1))
+        self._insert_event_for_time(ght.deleted_at + timedelta(seconds=1), hash)
         assert _query_for_issue(group2.id) == {group2.id: 1}
+
+    def test_organization_retention_respected(self):
+        base_time = datetime.utcnow()
+
+        self._insert_event_for_time(base_time - timedelta(minutes=1))
+        self._insert_event_for_time(base_time - timedelta(days=2))
+
+        def _get_event_count():
+            # attempt to query back 90 days
+            return snuba.query(
+                start=base_time - timedelta(days=90),
+                end=base_time + timedelta(days=1),
+                groupby=['project_id'],
+                filter_keys={
+                    'project_id': [self.project.id],
+                },
+            )
+
+        assert _get_event_count() == {self.project.id: 2}
+        with self.options({'system.event-retention-days': 1}):
```

Comment:
Using what's available in open source, but note this is richer in prod: https://github.com/getsentry/getsentry/blob/6043f6d5e75fadc426e3bec9dfea4ba80ab27b25/getsentry/quotas.py#L194-L202